src/biome/data/sources/datasource.py

```
@@ -143,7 +144,12 @@ def to_forward_dataframe(self) -> DataFrame:
         # This is strictly a shallow copy of the underlying computational graph
         forward_dataframe = self._df.copy()
 
-        if self.forward.label:
+        if (
+                not self.forward.label
+                or self.forward.label not in forward_dataframe.columns
```

I think this line could be a bit dangerous. Imagine i mistyped the label column name in the yml file of the datasource. Then my training would work without issues, taking the empty strings as the class label (see [how to deal with empty strings](https://gitlab.com/recognai-team/team/issues/31)).
So maybe i would just remove this line and let it fail on line 154, or raise a meaningful error message.

---

In deed, there is a big problem with `DataSource` class and forward mapping.

The current design is totally coupled to classification problems, but we can manage in future other kind of problems (NER, similarity), where we can apply some forward transformation.

We must avoid refers explicitly label manipulation in datasource, and keep in a more general and agnostic way